distribution/std-lib/Table/src/Table.enso

```diff
@@ -0,0 +1,95 @@
+from Base import all
+import Table.Io.Csv
+
+## Represents a column-oriented table data structure.
+type Table
+    type Table java_table
+
+    ## Returns a text containing an ASCII-art table displaying this data.
+
+       Arguments:
+         - show_rows: the number of initial rows that should be displayed.
+    display : Integer -> Text
+    display show_rows=10 =
+        cols = Vector (this.java_table.getColumns [])
+        col_names = cols.map (_.getName [])
+        col_vals = cols.map (_.getStorage [])
+        num_rows = this.java_table.nrows []
+        display_rows = min num_rows show_rows
+        rows = Vector.new display_rows row_num->
+            col_vals.map col->
+                if col.isNa [row_num] then "NA" else here.get_item_string col row_num
+        table = here.print_table col_names rows
+        if num_rows - display_rows <= 0 then table else
+            missing = '\n\u2026 and ' + (num_rows - display_rows).to_text + ' hidden rows.'
+            table + missing
+
+    ## Converts this table to a JSON structure.
+    to_json : Json
+    to_json =
+        col_jsons = Vector (this.java_table.getColumns []) . map here.column_to_json
+        cols_json = Json.Array col_jsons
+        fields = Map.singleton "columns" cols_json
+        Json.Object fields
+
+## PRIVATE
+
+   Keep this in sync with `org.enso.table.data.Storage.Type.LONG`
+storage_type_long = 1
+
+## PRIVATE
+
+   Keep this in sync with `org.enso.table.data.Storage.Type.DOUBLE`
+storage_type_double = 2
+
+## PRIVATE
+
+   Keep this in sync with `org.enso.table.data.Storage.Type.STRING`
+storage_type_string = 3
```

Comment:
Are these the only data types we support at present? Arrow supports [a whole host](https://arrow.apache.org/docs/python/data.html#type-metadata) of types, many of which are useful for us.

Reply:
Yes, this will be extended in the future.

Reply:
Please add a comment somewhere to this effect.

Reply:
I have no way of knowing that though! We extend it when we add arrow support, which is not currently scheduled at all.

Reply:
Moreover, I'm probably going to extend it soon-ish, when I add object storage